metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1684,6 +1698,19 @@ class MetalsLanguageServer(
     results
   }
 
+  /** Requesting semantic tokens for a whole file in order to highlight */
+  @JsonRequest("textDocument/semanticTokens/full")
+  def semanticTokensFull(
+      params: SemanticTokensParams
+  ): CompletableFuture[SemanticTokens] = {
+    CancelTokens.future { token =>
+      compilers.semanticTokens(
+        params,
+        token,
+      )
```

Comment:
```suggestion
      compilers.semanticTokens(
        params,
        token,
      ).map{ tokens =>
        if (tokens.isEmpty) null
        else tokens
      }
```
as noticed by @ckipp01 we need to return null in case of any issues or unsupported file format.